rbd/snapshot_nautilus.go

```diff
@@ -0,0 +1,68 @@
+// +build !luminous,!mimic
+//
+// Ceph Nautilus introduced rbd_get_parent() and deprecated rbd_get_parent_info().
+// Ceph Nautilus introduced rbd_list_children3() and deprecated rbd_list_children().
+
+package rbd
+
+// #cgo LDFLAGS: -lrbd
+// #include <rbd/librbd.h>
+// #include <errno.h>
+import "C"
+
+import (
+	"fmt"
+	"unsafe"
+)
+
+// int rbd_get_parent(rbd_image_t image, rbd_linked_image_spec_t *parent_image,
+//                    rbd_snap_spec_t *parent_snap)
+func (image *Image) GetParentInfo(p_pool, p_name, p_snapname []byte) error {
+	if err := image.validate(imageIsOpen); err != nil {
+		return err
+	}
+
+	parent_image := C.rbd_linked_image_spec_t{}
```

Comment:
For new code can we avoid underscores in var names? It'll cut down on the warnings from the checker tools.

Reply:
Done!